decidim-proposals/lib/decidim/content_parsers/proposal_parser.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentParsers
+    # A parser that searches mentions of Proposals in content.
+    #
+    # This parser accepts two ways for linking Proposals.
+    # - Using a standard url starting with http or https.
+    # - With a word starting with `~` and digits afterwards will be considered a possible mentioned proposal.
+    # For example `~1234`, but no `~ 1234`.
+    #
+    # Also fills a `Metadata#linked_proposals` attribute.
+    #
+    # @see BaseParser Examples of how to use a content parser
+    class ProposalParser < BaseParser
+      # Class used as a container for metadata
+      #
+      # @!attribute linked_proposals
+      #   @return [Array] an array of Decidim::Proposals::Proposal mentioned in content
+      Metadata = Struct.new(:linked_proposals)
+
+      # Matches a URL
+      URL_REGEX_PART1 = '(?i)\b((?:[a-z][\w-]+:(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}\/)(?:[^\s()<>]+|'
+      URL_REGEX_PART2 = '\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:\'".,<>?«»“”‘’]))'
+      URL_REGEX = /#{URL_REGEX_PART1}#{URL_REGEX_PART2}/i
+      # Matches a mentioned Proposal ID (~(d)+ expression)
+      ID_REGEX = /~(\d+)/
+
+      def initialize(content, context)
+        super
+        @metadata = Metadata.new([])
+      end
+
+      # Replaces found mentions matching an existing
+      # Proposal with a global id for that Proposal. Other mentions found that doesn't
+      # match an existing Proposal are returned as they are.
+      #
+      # @return [String] the content with the valid mentions replaced by a global id.
+      def rewrite
+        rewrited_content = parse_for_urls(content)
+        parse_for_ids(rewrited_content)
+      end
+
+      # (see BaseParser#metadata)
+      attr_reader :metadata
+
+      #-------------------------------------------------
+
+      private
+
+      #-------------------------------------------------
```

Comment:
This too

Reply:
I'm removing it but let's talk about it next time we meet!